Author: Jordi Mallach <jordi@debian.org>
Description: avoid compiling linenoise, that includes the non-free file
 ConvertUTF.cpp.

--- a/makefile
+++ b/makefile
@@ -821,6 +821,10 @@
 ifdef SANITIZE
 PARAMS += --SANITIZE='$(SANITIZE)'
 endif
+
+ifdef NO_USE_LINENOISE
+PARAMS += --NO_USE_LINENOISE=1
+endif
 #-------------------------------------------------
 # All scripts
 #-------------------------------------------------
--- a/scripts/genie.lua
+++ b/scripts/genie.lua
@@ -393,6 +393,11 @@
 }
 
 newoption {
+	trigger = "NO_USE_LINENOISE",
+	description = "Avoid using linenoise for licensing reasons.",
+}
+
+newoption {
 	trigger = "PROJECT",
 	description = "Select projects to be built. Will look into project folder for files.",
 }
--- a/scripts/src/3rdparty.lua
+++ b/scripts/src/3rdparty.lua
@@ -911,7 +911,7 @@
 	includedirs {
 		MAME_DIR .. "3rdparty",
 	}
-if (_OPTIONS["osd"] ~= "uwp") then
+if (_OPTIONS["osd"] ~= "uwp") and (not _OPTIONS["NO_USE_LINENOISE"]) then
 	includedirs {
 		MAME_DIR .. "3rdparty/linenoise",
 	}
@@ -934,7 +934,7 @@
 		MAME_DIR .. "3rdparty/lua-zlib/lua_zlib.c",
 		MAME_DIR .. "3rdparty/luafilesystem/src/lfs.c",
 	}
-if (_OPTIONS["osd"] == "uwp") then
+if (_OPTIONS["osd"] == "uwp") or (_OPTIONS["NO_USE_LINENOISE"]) then
 	files {
 		MAME_DIR .. "3rdparty/lua-linenoise/linenoise_none.c",
 	}
@@ -2035,7 +2035,7 @@
 --------------------------------------------------
 -- linenoise library
 --------------------------------------------------
-if (_OPTIONS["osd"] ~= "uwp") then
+if (_OPTIONS["osd"] ~= "uwp") and (not _OPTIONS["NO_USE_LINENOISE"]) then
 project "linenoise"
 	uuid "7320ffc8-2748-4add-8864-ae29b72a8511"
 	kind (LIBTYPE)
--- a/scripts/src/main.lua
+++ b/scripts/src/main.lua
@@ -264,7 +264,7 @@
 		ext_lib("lua"),
 		"lualibs",
 	}
-if (_OPTIONS["osd"] ~= "uwp") then
+if (_OPTIONS["osd"] ~= "uwp") and (not _OPTIONS["NO_USE_LINENOISE"]) then
 	links {
 		"linenoise",
 	}
